pkg/rbac/parser.go

```diff
@@ -146,69 +147,110 @@ func (Generator) RegisterMarkers(into *markers.Registry) error {
 	return nil
 }
 
-// GenerateClusterRole generates a rbacv1.ClusterRole object
-func GenerateClusterRole(ctx *genall.GenerationContext, roleName string) (*rbacv1.ClusterRole, error) {
-	rules := make(map[ruleKey]*Rule)
+// GenerateRoles generate a slice of objs representing either a ClusterRole or a Role object
+// The order of the objs in the returned slice is stable and determined by their namespaces.
+func GenerateRoles(ctx *genall.GenerationContext, roleName string) ([]interface{}, error) {
```

Comment:
I would keep this as it. Because `WriteYAML expects `...interface{}`:
```
func (g GenerationContext) WriteYAML(itemPath string, objs ...interface{}) error
```